services/lighthouse/src/main.rs

```diff
@@ -25,17 +34,39 @@ enum LighthouseError {
     KeyDecodeError(#[from] base64::DecodeError),
     #[error("failed to query database")]
     QueryError(#[from] lighthouse_prisma::client::queries::QueryError),
+    #[error("failed to parse endpoint ip:port pair")]
+    EndpointParseError(#[from] AddrParseError),
+    #[error("failed to use public key")]
+    PubkeyError(#[from] pgp::errors::Error),
+    #[error("failed to verify signature")]
+    SigError,
+    #[error("no such ID")]
+    InvalidID,
+    #[error("invalid fingerprint")]
+    InvalidFingerprint(#[from] hex::FromHexError),
+}
+
+#[derive(Serialize)]
+struct ServerError {
+    sorry: String,
+    error: String,
 }
 
 impl IntoResponse for LighthouseError {
     fn into_response(self) -> Response {
-        todo!()
+        let mut resp = Json(ServerError {
+            sorry: "oopsies :3".to_string(),
+            error: format!("{:?}", self), // self.to_string()
+        })
+        .into_response();
+        *resp.status_mut() = StatusCode::BAD_REQUEST;
+        resp
     }
 }
 
 /// Defines the context for requests.
 struct LighthouseCtx {
-    db: RwLock<PrismaClient>,
+    db: Arc<RwLock<PrismaClient>>,
 }
```

Comment:
Arcing the fields as opposed to the entire context?

Reply:
I think I kept the Arc<LighthouseCtx> from the original code. The Arc here I think should be ok

Reply:
> I think I kept the Arc from the original code. The Arc here I think should be ok

Oh, is it because you move a copy of the Prisma client inside of one of the worker threads? That would make sense, although that vs moving a copy of Ctx?

Reply:
Yeah I directly use the client rather than the Ctx itself, because I wasn't sure if the web handlers using the Ctx in their Extension had some requirement or something.